packages/apps/reputation-oracle/server/src/modules/exchange/mexc-exchange.client.ts

```diff
@@ -0,0 +1,98 @@
+import { createHmac } from 'node:crypto';
+
+import { type SupportedExchange } from '@/common/constants';
+import logger from '@/logger';
+
+import { ExchangeApiClientError } from './errors';
+import type {
+  ExchangeClient,
+  ExchangeClientCredentials,
+  ExchangeClientOptions,
+} from './types';
+import { fetchWithHandling } from './utils';
+
+const MEXC_API_BASE_URL = 'https://api.mexc.com/api/v3';
+
+export class MexcExchangeClient implements ExchangeClient {
+  readonly id: SupportedExchange = 'mexc';
+  private readonly apiKey: string;
+  private readonly secretKey: string;
+  private readonly timeoutMs?: number;
+  private readonly logger = logger.child({
+    context: MexcExchangeClient.name,
+    exchange: this.id,
+  });
+  readonly recvWindow = 5000;
+
+  constructor(
+    creds: ExchangeClientCredentials,
+    options?: ExchangeClientOptions,
+  ) {
+    if (!creds?.apiKey || !creds?.secretKey) {
+      throw new ExchangeApiClientError('Incomplete credentials for MEXC');
+    }
+    this.apiKey = creds.apiKey;
+    this.secretKey = creds.secretKey;
+    this.timeoutMs = options?.timeoutMs;
+  }
+
+  private signQuery(query: string): string {
+    return createHmac('sha256', this.secretKey).update(query).digest('hex');
+  }
+
+  async checkRequiredAccess(): Promise<boolean> {
+    const path = '/account';
+    const timestamp = Date.now();
+    const query = `timestamp=${timestamp}&recvWindow=${this.recvWindow}`;
+    const signature = this.signQuery(query);
+    const url = `${MEXC_API_BASE_URL}${path}?${query}&signature=${signature}`;
+
+    const res = await fetchWithHandling(
+      this.id,
+      url,
+      { 'X-MEXC-APIKEY': this.apiKey },
+      this.logger,
+      this.timeoutMs,
+    );
+    if (res.ok) return true;
+    this.logger.debug('MEXC access check failed', {
+      status: res.status,
+      statusText: res.statusText,
+    });
+    return false;
+  }
+
+  async getAccountBalance(asset: string): Promise<number> {
+    const path = '/account';
+    const timestamp = Date.now();
+    const query = `timestamp=${timestamp}&recvWindow=${this.recvWindow}`;
+    const signature = this.signQuery(query);
+    const url = `${MEXC_API_BASE_URL}${path}?${query}&signature=${signature}`;
+
+    const res = await fetchWithHandling(
+      this.id,
+      url,
+      { 'X-MEXC-APIKEY': this.apiKey },
+      this.logger,
+      this.timeoutMs,
+    );
+    if (!res.ok) {
+      this.logger.warn('MEXC balance fetch failed', {
```

Comment:
```suggestion
      this.logger.debug('MEXC balance fetch failed', {
```
I'm not sure that we want to log any non-2xx code.
Let's log only errors in `fetchWithHandling` (so we get major code errors or sensible network errors only), if somebody doesn't have proper access or exchange is 500/502 it will be just noise for us if we log it other than debug